### DISCUSSION TITLE
Model attribute hydration bug when attribute uses collection cast

### DISCUSSION THREAD

**Discussion by gdebrauwer** [livewire/livewire#5502] [7. Bugs (archived)]
When a model attribute uses the `AsCollection` cast (or a similar custom cast), then updating the value of that attribute results in a mix of the original values and the updated value

## Reproduce

1. Add array attribute to 'User' model and define cast
```php
class User
{
    public $fillable = [
        // ...
        'sports',
    ];

    public $casts = [
        // ...
        'sports' => \Illuminate\Database\Eloquent\Casts\AsCollection::class,
    ];
}
```
```php
public function up() : void
{
        Schema::create('users', function (Blueprint $table) {
            // ...
            $table->json('sports')->nullable();
        });
}
```
2. Create following livewire component
```php
class UpdateSportsOfUser extends Component
{
    public User $user;

    public function mount() : void
    {
        $this->user = User::factory()->create(['sports' => ['padel','tennis']);
    }

    public function update() : void
    {
        dump($this->user->sports);
    }

    protected function rules() : array
    {
        return [
            'user.sports' => [
                 'array',
            ],
             'user.sports.*' => [
                 'string',
            ]
        ];
    }
}
```
```blade
<form wire:submit.prevent="update" wire:init="$set('user.sports', ['squash'])">
    <button type="submit">update</button>
</form>
```
3. Click the "update" button. The dump will show `['squash', 'tennis']` instead of expected `['squash']`

## Possible cause

Based on looking through the livewire source code, I think this bug is caused by the fact that the `setDirtyData` method in `HydratePublicProperties` middleware checks if the $existingData is an array, which does not work if it is a collection because of the `AsCollection` cast. https://github.com/livewire/livewire/blob/4cc5dedaab1e9512efb4d528fde67df98e9b465a/src/HydrationMiddleware/HydratePublicProperties.php#L193-L203

**Comment by joshhanley** [comment]
@gdebrauwer yeah looks like a bug to me. PR's are welcome! 🙂
